### PR TITLE
Added site recovery and srm nodes resource.

### PR DESCRIPTION
**Pull request by saloneerege**

### CODE REVIEW

File: vmc/resource_vmc_site_recovery.go [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+				ForceNew:    true,
+				Required:    true,
+				Description: "SDDC identifier",
+			},
+			"srm_extension_key_suffix": {
+				Type:         schema.TypeString,
+				ForceNew:     true,
+				Optional:     true,
+				ValidateFunc: validation.StringLenBetween(0, 13),
+				Description:  "Custom extension key suffix for SRM. If not specified, default extension key will be used. The custom extension suffix must contain 13 characters or less, be composed of letters, numbers, ., - characters only. The extension suffix must begin and end with a letter or number. The suffix is appended to com.vmware.vcDr- to form the full extension key",
```
Comment (shylasrinivas) [inline vmc/resource_vmc_site_recovery.go]: can we make the description shorter?

File: vmc/resource_vmc_site_recovery.go [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+			},
+			"site_recovery_state": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Site recovery state. Possible values are: ACTIVATED, ACTIVATING, CANCELED, DEACTIVATED, DEACTIVATING, DELETED, FAILED",
```
Comment (shylasrinivas) [inline vmc/resource_vmc_site_recovery.go:49]: possible values state values can be stored in a string[] and validated here

Comment (saloneerege) [reply vmc/resource_vmc_site_recovery.go:49]: This is a computed value. They are returned as a part of the response on GET. The possible values are only being mentioned in the description for reference.

File: vmc/resource_vmc_site_recovery.go (hunk 1 above)
Comment (saloneerege) [reply vmc/resource_vmc_site_recovery.go]: Sure. I will update it.

File: vmc/resource_vmc_site_recovery.go [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+
+}
+
+func resourceSiteRecoveryRead(d *schema.ResourceData, m interface{}) error {
+	connector := (m.(*ConnectorWrapper)).Connector
+	sddcID := d.Id()
+	orgID := (m.(*ConnectorWrapper)).OrgID
+	siteRecoveryClient := draas.NewDefaultSiteRecoveryClient(connector)
+	siteRecovery, err := siteRecoveryClient.Get(orgID, sddcID)
+	if err != nil {
```
Comment (shylasrinivas) [inline vmc/resource_vmc_site_recovery.go:127]: we should also check for resource not found error. Usually resource is re-created if its not found due to force new param, not sure what the use case here is.

File: vmc/resource_vmc_site_recovery.go [hunk 4]
```diff
@@ -0,0 +1,205 @@
...
+	})
+}
+
+func resourceSiteRecoveryUpdate(d *schema.ResourceData, m interface{}) error {
+	if d.HasChange("srm_extension_key_suffix") {
+		err := resource.NonRetryableError(resourceSiteRecoveryDelete(d, m))
+		if err != nil {
+			return fmt.Errorf("Error while deactivating site recovery: %v", err)
+		}
+		time.Sleep(15 * time.Minute)
```
Comment (shylasrinivas) [inline vmc/resource_vmc_site_recovery.go:213]: can you add a comment regarding the sleep?

**Review by shylasrinivas — APPROVED** [review]

File: vmc/resource_vmc_site_recovery.go (hunk 3 above)
Comment (saloneerege) [reply vmc/resource_vmc_site_recovery.go:127]: Done

File: vmc/resource_vmc_site_recovery.go (hunk 4 above)
Comment (saloneerege) [reply vmc/resource_vmc_site_recovery.go:213]: Done

File: vmc/resource_vmc_srm_nodes.go [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+	"github.com/vmware/vsphere-automation-sdk-go/services/vmc/draas"
+	"github.com/vmware/vsphere-automation-sdk-go/services/vmc/draas/model"
+)
+
+func resourceSRMNodes() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceSRMNodesCreate,
+		Read:   resourceSRMNodesRead,
+		Delete: resourceSRMNodesDelete,
+		//Update: resourceSRMNodesUpdate,
```
Comment (shylasrinivas) [inline vmc/resource_vmc_srm_nodes.go]: this line can be deleted if its not going to be supported

File: vmc/resource_vmc_srm_nodes.go [hunk 6]
```diff
@@ -0,0 +1,159 @@
...
+			log.Printf("SRM information for SDDC with ID %s not found", sddcID)
+			d.SetId("")
+			return fmt.Errorf("SRM information for SDDC with ID %s not found", sddcID)
+		}
+		return fmt.Errorf("Error while getting SRM instance information for SDDC with ID %s : %v", sddcID, err)
+	}
+	srmExtensionKey := d.Get("srm_node_extension_key_suffix").(string)
+	srm_node := map[string]string{}
+	var i int
+	for i = 0; i < len(siteRecovery.SrmNodes); i++ {
```
Comment (shylasrinivas) [inline vmc/resource_vmc_srm_nodes.go]: int declaration on line 119 can be removed by changing for to "for i := 0.."

File: vmc/resource_vmc_srm_nodes.go (hunk 5 above)
Comment (saloneerege) [reply vmc/resource_vmc_srm_nodes.go]: I will be adding that support soon. Waiting for a clarification from DRaaS team. But I will remove the line for now.

File: vmc/resource_vmc_srm_nodes.go (hunk 6 above)
Comment (saloneerege) [reply vmc/resource_vmc_srm_nodes.go]: Done

**Review by shylasrinivas — APPROVED** [review]

**Pull request merged**